x-pack/filebeat/input/azureblobstorage/input.go

```diff
@@ -0,0 +1,150 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build !aix
+// +build !aix
+
+package azureblobstorage
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	v2 "github.com/elastic/beats/v7/filebeat/input/v2"
+	cursor "github.com/elastic/beats/v7/filebeat/input/v2/input-cursor"
+	"github.com/elastic/beats/v7/libbeat/feature"
+	"github.com/elastic/beats/v7/x-pack/filebeat/input/azureblobstorage/state"
+	"github.com/elastic/beats/v7/x-pack/filebeat/input/azureblobstorage/types"
+	conf "github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+type azurebsInput struct {
+	config     config
+	serviceURL string
+}
+
+const (
+	inputName string = "azureblobstorage"
+)
+
+func Plugin(log *logp.Logger, store cursor.StateStore) v2.Plugin {
+	return v2.Plugin{
+		Name:       inputName,
+		Stability:  feature.Experimental,
+		Deprecated: false,
+		Info:       "Azure Blob Storage logs",
+		Doc:        "Collect logs from Azure Blob Storage Service",
+		Manager: &cursor.InputManager{
+			Logger:     log,
+			StateStore: store,
+			Type:       inputName,
+			Configure:  configure,
+		},
+	}
+}
+
+func configure(cfg *conf.C) ([]cursor.Source, cursor.Input, error) {
+	config := defaultConfig()
+	if err := cfg.Unpack(&config); err != nil {
+		return nil, nil, err
+	}
+
+	var sources []cursor.Source
+	for _, c := range config.Containers {
+		container := tryOverrideOrDefault(config, c)
+		sources = append(sources, &types.Source{
+			AccountName:   config.AccountName,
+			ContainerName: c.Name,
+			MaxWorkers:    *container.MaxWorkers,
+			Poll:          *container.Poll,
+			PollInterval:  *container.PollInterval,
+		})
+	}
+
+	url := fmt.Sprintf("https://%s.blob.core.windows.net/", config.AccountName)
+	return sources, &azurebsInput{config: config, serviceURL: url}, nil
+}
+
+// tryOverrideOrDefault , overrides global values with local
+// container level values present. If both global & local values
+// are absent , assigns default values
+func tryOverrideOrDefault(cfg config, c container) container {
+
```

Comment:
nit: Will you please [`gofumpt -w .`](https://github.com/mvdan/gofumpt) this package (I'm looking at the "No empty lines around function bodies" rule 😄 ).

Reply:
sure